src/platforms/posix/drivers/df_ak8963_wrapper/df_ak8963_wrapper.cpp

```diff
@@ -279,9 +279,9 @@ int DfAK8963Wrapper::_publish(struct mag_sensor_data &data)
 	mag_report.timestamp = hrt_absolute_time();
 
 	// TODO: remove these (or get the values)
-	mag_report.x_raw = NAN;
-	mag_report.y_raw = NAN;
-	mag_report.z_raw = NAN;
+	mag_report.x_raw = 0;
+	mag_report.y_raw = 0;
+	mag_report.z_raw = 0;
```

Comment:
Don't cast to int, and 0 is as wrong as -2147483648.

If you have a float that is NaN, you can at least nicely catch it with `is_nan` or `!is_finite`.